src/main/java/de/hpi/bpt/argos/persistence/model/parsing/BackboneDataParserImpl.java

```diff
@@ -0,0 +1,431 @@
+	 */
+	protected void saveCurrentProduct() {
+
+		if (currentProduct == null) {
+			return;
+		}
+
+		logger.info(String.format("added product '%1$s' with '%2$d' configurations in family '%3$s'", currentProduct.getName(), currentProduct
+				.getProductConfigurations().size(), currentProduct.getProductFamily().getName()));
+		entityManager.updateEntity(currentProduct, ProductEndpoint.getProductUri(currentProduct.getId()));
+
+		resetCurrentEntities();
+	}
+
+	/**
+	 * This method resets all temporary variables.
+	 */
+	protected void resetCurrentEntities() {
+
+		tempCurrentProductExternalId = "";
+		tempCurrentProductFamilyId = "";
+		tempCurrentCauseDescription = "";
+		tempCurrentProductName = "";
+		tempCurrentCauseCode = "";
+		tempCurrentErrorDisplayCode = "";
+		tempCurrentCodingPlug = "";
+		currentProduct = null;
+		currentErrorType = null;
+		currentProductConfiguration = null;
+	}
+
+	/**
+	 * This method checks, whether a error type is already present in the current product.
+	 * @param errorType - the error type to check
+	 * @return - true, if the error type is already present
+	 */
+	protected boolean currentProductConfigurationHasErrorType(ErrorType errorType) {
+		if (currentProductConfiguration == null) {
+			return false;
+		}
+
+		for (ErrorType type : currentProductConfiguration.getErrorTypes()) {
+			if (type.getErrorTypeId().equalsIgnoreCase(errorType.getErrorTypeId())) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+    /**
+     * Logs an exception on log level trace.
+     * @param e - exception to log
+     */
+	private void logTrace(Exception e) {
+        logger.trace("Reason: ", e);
+    }
+  
+	/**
+	 * This method logs the statistics about the imported entities.
+	 */
+	protected void logStatistics() {
+		logger.info(String.format("finished importing data from '%1$s' in '%2$s'. "
+				+ "imported %3$d products, "
+				+ "%4$d productConfigurations, "
+				+ "%5$d errorTypes and "
+				+ "%6$d errorCauses.",
+				fileName,
+				getTimeSinceStart(),
+				productsImported,
+				productConfigurationsImported,
+				errorTypesImported,
+				errorCausesImported));
+	}
+
+	/**
+	 * This method calculates the passed time since the import process started and returns it as a string.
+	 * @return - the passed time since start of the import process as string
+	 */
+	protected String getTimeSinceStart() {
+
+		long passedTimeInSeconds = ((new Date()).getTime() - importStartTime.getTime()) / 1000;
+		long passedTime[] = new long[] { 0, 0, 0, 0 };
+
+		// sec
+		passedTime[3] = passedTimeInSeconds >= 60 ? passedTimeInSeconds % 60 : passedTimeInSeconds;
+		// min
+		passedTime[2] = (passedTimeInSeconds = passedTimeInSeconds / 60) >= 60 ? passedTimeInSeconds % 60 : passedTimeInSeconds;
+		// hour
+		passedTime[1] = (passedTimeInSeconds = passedTimeInSeconds / 60) >= 24 ? passedTimeInSeconds % 24 : passedTimeInSeconds;
+		// day
+		passedTime[0] = passedTimeInSeconds / 24;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Assign this magic number 24 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS109)